finality-aleph/src/sync/handler/mod.rs

```diff
@@ -708,6 +718,137 @@ mod tests {
         top
     }
 
+    #[tokio::test]
+    async fn finalizes_with_justification_gaps() {
+        let (mut handler, _backend, mut notifier, genesis) = setup();
+        let mut bottom = genesis;
+        let peer_id = 0;
+        for session in 0.. {
+            let top = SESSION_BOUNDARY_INFO.last_block_of_session(SessionId(session));
+            let branch = grow_light_branch_till(&mut handler, &bottom, &top, peer_id);
+            bottom = branch.last().expect("should not be empty").id();
+            // import blocks
+            let response_items = branch_response(branch.clone(), true, true, false);
```

Comment:
Ugh, three bools as arguments... could you make this into a single `enum` instead? Or, if you really need all possible combinations, a small struct which would make this more readable.

Reply:
struct it is